server/api/resolvers/requestGroupResolvers.ts

```diff
@@ -0,0 +1,139 @@
+    requestGroups: async (_, __, ___): Promise<Array<RequestGroupInterface>> => {
+        return RequestGroup.find().exec()
+    },
+    requestGroupsFilter: async (_, { filter, options }, ___): Promise<Array<RequestGroupInterface>> => {
+        return RequestGroup.find().exec()
+    },
+}
+
+const requestGroupMutationResolvers = {
+    createRequestGroup: async (_, { requestGroup }, { authenticateUser }): Promise<RequestGroupInterface> => {
+        return authenticateUser().then(async () => { 
+            const newRequestGroup = new RequestType({...requestGroup})
+            const createdRequestGroup = await newRequestGroup.save()
+
+            return createdRequestGroup
+        })
+    },
+    updateRequestGroup: async (_, { requestGroup }, { authenticateUser }): Promise<RequestGroupInterface> => {
+        return authenticateUser().then(async () => {
+            const oldRequestGroup = await RequestGroup.findById(requestGroup._id)
+            const modifiedRequestGroup = new RequestType({...oldRequestGroup, ...requestGroup})
+
+            return modifiedRequestGroup.save()
+        })
+    },
+    deleteRequestGroup: async (_, { _id }, { authenticateUser }): Promise<RequestGroupInterface> => {
+        return authenticateUser().then(async () => {
+            const requestGroup = await RequestGroup.findById(_id)
+            requestGroup.deletedAt = new Date()
+            return requestGroup.save()
+        })
+    },
+}
+
+const requestGroupResolvers = {
+    requestTypes: async (parent, __, ___): Promise<Array<RequestTypeInterface>> => {
+        return parent.requestTypes.map((requestTypeEmbedding) => {
+            return RequestType.findById(requestTypeEmbedding._id)
+        })
+    },
+    deleted: (parent, __, ___): boolean => {
+        return parent.deletedAt !== undefined
+    },
+    countOpenRequests: async (parent, __, ___): Promise<number> => {
+        const countOpenRequestsPerType: Array<number> = await Promise.all(parent.requestTypes.map(async (requestTypeEmbedding) => {
+            const requestType = await RequestType.findById(requestTypeEmbedding._id)
+            return filterOpenRequestEmbeddings(requestType.requests).length
+        }))
+
+        return countOpenRequestsPerType.reduce((total : number, openRequestsInRequestType : number) => total + openRequestsInRequestType, 0)
+    },
+    nextRequest: async (parent, __, { authenticateUser }): Promise<RequestInterface> => {
+        return authenticateUser().then(async () => {
+            return sessionHandler(async (session) => {
+                const nextRequestEmbedding = await nextRequestEmbeddingForRequestGroup(parent, session)
+
+                if (nextRequestEmbedding === null) {
+                    return null
+                }
+
+                return Request.findById(nextRequestEmbedding._id).session(session)
+            })
+        })
+    },
+    nextRecipient: async (parent, __, { authenticateUser }): Promise<ClientInterface> => {
```

Comment:
sorry but what does nextRecipient mean for a requestGroup?

Reply:
The Client whose Request is next in line (FIFO)